client/my-sites/controller.js

```diff
@@ -201,14 +201,13 @@ function isPathAllowedForDomainOnlySite( path, slug, primaryDomain ) {
 		);
 	}
 
-	const otherPaths = [ `/checkout/${ slug }` ];
```

Comment:
The renewal links are in the `/checkout/domain_map:{:domain}/renew/{:subscriptionId}/{:slug}` format, so neither of those rules allowed for it.